File: problem04/spec/p04_spec.js
var assert = require('assert');
var app = require('../p04.js');

describe('#isPalindrome', function() {
  it('should return true if string is a palindrome', function() {
    var test = 'anna';
    assert.equal(true, app.isPalindrome(test));
  });

  it('should return false if string is not a palindrome', function() {
    var test = 'eddie';
    assert.equal(false, app.isPalindrome(test));
  });
});

describe('#solution', function() {
  it('should return the largest 3 digit palindrome', function() {
    var result = 906609;
    assert.equal(result, app.solution());
  });
});

File: problem04/spec/p04_spec.rb
require_relative '../p04.rb'

describe '#is_palindrome?' do
  it 'should return true if string is a palindrome' do
    string = 'anna'
    expect(is_palindrome?(string)).to eq(true)
  end

  it 'should return false is string is not a palindrome' do
    string = 'eddie'
    expect(is_palindrome?(string)).to eq(false)
  end
end

describe '#solution' do
  it 'should return the largest 3 digit palindrome' do
    result = 906609
    expect(solution).to eq(result)
  end
end
